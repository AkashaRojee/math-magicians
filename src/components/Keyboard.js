/* eslint-disable */

import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import Key from './Key';
import KeyStyle from '../objects/KeyStyle';

const Keyboard = (props) => {
  const keyStyles = {
    clear: 'bg-purple',
    symbol: 'bg-purple-light',
    number: 'bg-purple-dark',
    equal: 'bg-cyan',
  };

  const keyValues = [
    new KeyStyle('AC', keyStyles.clear),
    new KeyStyle('+/-', keyStyles.symbol),
    new KeyStyle('%', keyStyles.symbol),
    new KeyStyle('/', keyStyles.symbol),
    new KeyStyle('7', keyStyles.number),
    new KeyStyle('8', keyStyles.number),
    new KeyStyle('9', keyStyles.number),
    new KeyStyle('x', keyStyles.symbol),
    new KeyStyle('4', keyStyles.number),
    new KeyStyle('5', keyStyles.number),
    new KeyStyle('6', keyStyles.number),
    new KeyStyle('-', keyStyles.symbol),
    new KeyStyle('1', keyStyles.number),
    new KeyStyle('2', keyStyles.number),
    new KeyStyle('3', keyStyles.number),
    new KeyStyle('+', keyStyles.symbol),
    new KeyStyle('0', `span-two-col ${keyStyles.number}`),
    new KeyStyle('.', keyStyles.number),
    new KeyStyle('=', keyStyles.equal),
  ];

  const handleClick = (keyValue) => {
    props.onKeyboardClick(keyValue);
  };

  const renderOutput = useMemo(() => (
    <div className="keyboard">
      {keyValues.map(({keyValue, className}) => (
        <Key
          value={keyValue}
          key={keyValue}
          className={className}
          onKeyClick={(keyValue) => handleClick(keyValue)}
        />
      ))}
    </div>
  ));

  return renderOutput;
};

Keyboard.propTypes = {
  onKeyboardClick: PropTypes.func.isRequired,
};

export default Keyboard;
