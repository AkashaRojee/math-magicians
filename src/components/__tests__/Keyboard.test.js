import renderer from 'react-test-renderer';
import Keyboard from '../Keyboard';

test('Keyboard renders correctly', () => {
  const handleClick = jest.fn();

  const tree = renderer
    .create(<Keyboard onKeyboardClick={() => handleClick()} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
