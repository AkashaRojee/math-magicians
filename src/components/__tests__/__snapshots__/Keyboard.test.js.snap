// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyboard renders correctly 1`] = `
<div
  className="keyboard"
>
  <button
    className="bg-purple"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="bg-purple-light"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="bg-purple-light"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="bg-purple-light"
    onClick={[Function]}
    type="button"
  >
    /
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="bg-purple-light"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="bg-purple-light"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="bg-purple-light"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="span-two-col bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="bg-purple-dark"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="bg-cyan"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
